### ISSUE TITLE
DeprecationWarnings in axis.py after updating NumPy

### ISSUE THREAD

**Issue by halsafar** [enthought/chaco#266]
We updated NumPy in our project virtual environment and now we get a ton of terminal spam in the form of:

```
python2.7/site-packages/chaco-4.5.0-py2.7-linux-x86_64.egg/chaco/axis.py:498:
 DeprecationWarning: elementwise comparison failed; this will raise the
 error in the future.
   for tickpos in mapped_tick_positions]))
python2.7/site-packages/chaco-4.5.0-py2.7-linux-x86_64.egg/chaco/axis.py:499:
 DeprecationWarning: elementwise comparison failed; this will raise the
 error in the future.
   self._tick_label_list = tick_list
```

We updated chaco to the newest stable and we still get the deprecation warnings.

**Comment by jonathanrocher** [comment]
Thanks for reporting the issue. What version of NumPy did you update to? Until we look into this, if you need to turn off warnings, it is possible to filter them out of your terminal output using https://docs.python.org/2/library/warnings.html .

**Comment by corranwebster** [comment]
This is almost certainly related to #251.

**Comment by corranwebster** [comment]
To add a little color, this is a change introduced in NumPy 1.9.0 that changes the way that NumPy compares arrays to objects to more consistently use elementwise comparison. See http://docs.scipy.org/doc/numpy/release.html#object-array-equality-comparisons

In particular `arr == None` will behave differently in the future, and this comparison is particularly prevalent in Traits-based code that interacts with NumPy.

At the present time these are just warnings and NumPy's behaviour is unchanged, so Chaco should still be working correctly. As @jonathanrocher mentioned above, it is possible to filter these warnings out using the standard library's warning module (or even print a particular warning just once.  In fact, it is a little surprising that you are seeing DeprecationWarnings at all, since they are filtered out by default in Python 2.7.

We've already made the appropriate changes in Traits that will allow us to resolve this issue, and I would expect that we will be resolving it in Chaco and other libraries in the near term.

**Comment by halsafar** [comment]
Thanks for the movement on this.

We updated straight to NumPy 1.9.1 when we noticed this issue.  Although based on previous comments it appears that it was NumPy 1.9.0 that started the issue.

**Comment by halsafar** [comment]
Any movement on this?

**Comment by rkern** [comment]
Fixed by #350
